src/mem/old_api.cc

```diff
@@ -432,22 +411,29 @@ Mem::Init(void)
     /**
      * Then initialize all pools.
      * \par
-     * Starting with generic 2kB - 64kB buffr pools, then specific object types.
+     * Starting with generic buffer pools, then specific object types.
      * \par
      * It does not hurt much to have a lot of pools since sizeof(MemPool) is
      * small; someday we will figure out what to do with all the entries here
      * that are never used or used only once; perhaps we should simply use
      * malloc() for those? @?@
      */
+    memDataInit(MEM_32B_BUF, "32B Buffer", 32, 10, false);
+    memDataInit(MEM_64B_BUF, "64B Buffer", 64, 10, false);
+    memDataInit(MEM_128B_BUF, "128B Buffer", 128, 10, false);
+    memDataInit(MEM_256B_BUF, "256B Buffer", 256, 10, false);
+    memDataInit(MEM_512B_BUF, "512B Buffer", 512, 10, false);
+    memDataInit(MEM_1K_BUF, "1K Buffer", 1024, 10, false);
     memDataInit(MEM_2K_BUF, "2K Buffer", 2048, 10, false);
     memDataInit(MEM_4K_BUF, "4K Buffer", 4096, 10, false);
     memDataInit(MEM_8K_BUF, "8K Buffer", 8192, 10, false);
     memDataInit(MEM_16K_BUF, "16K Buffer", 16384, 10, false);
     memDataInit(MEM_32K_BUF, "32K Buffer", 32768, 10, false);
     memDataInit(MEM_64K_BUF, "64K Buffer", 65536, 10, false);
-    memDataInit(MEM_DREAD_CTRL, "dread_ctrl", sizeof(dread_ctrl), 0);
-    memDataInit(MEM_DWRITE_Q, "dwrite_q", sizeof(dwrite_q), 0);
-    memDataInit(MEM_MD5_DIGEST, "MD5 digest", SQUID_MD5_DIGEST_LENGTH, 0);
+    // TODO: make these objects use MEMPROXY_CLASS() and drop the doZero parameter
```

Comment:
I disagree with the suggested MEMPROXY_CLASS() solution (because, in part, MD5_DIGEST-using objects should not become MEMPROXY_CLASS), but I agree with the _essential_ parts of this TODO. Here is a variant that may avoid disagreements:

```suggestion
    // TODO: Carefully stop zeroing these objects memory and drop the doZero parameter
```

I do _not_ insist on the comment using the word "Carefully". FWIW, before suggesting that word, I have looked at how these buffers are used right now. Here are my draft notes:

MEM_DREAD_CTRL zeroing is a waste because the corresponding dread_ctrl object fields are fully initialized upon allocation and uninitialized padding is irrelevant in this context. The type should be upgraded to a pooled C++ class. I recommend doing that in a dedicated PR.

MEM_DWRITE_Q zeroing may be essential in the current (arguably buggy) code because one of the dwrite_q users does _not_ initialize the dwrite_q::file_offset field -- there is a corresponding XXX in diskCombineWrites(). The type should be upgraded to a pooled C++ class, but resolving that XXX is not trivial and deserves a dedicated PR.

MEM_MD5_DIGEST zeroing is a waste because the buffer is immediately filled with SQUID_MD5_DIGEST_LENGTH bytes. Most likely, MEM_MD5_DIGEST-based cache_key type should be changed from a buffer to two uint64_t integers, but a clean upgrade requires a lot of work and warrants a dedicated PR.